server/app/org/scalaexercises/exercises/controllers/ApplicationController.scala

```diff
@@ -65,99 +77,126 @@ class ApplicationController(cache: CacheApi)(
     }
   }
 
-  def index = Secure(Action.async { implicit request ⇒
-
-    val ops = for {
-      authorize ← githubOps.getAuthorizeUrl(ConfigUtils.githubAuthId, ConfigUtils.callbackUrl)
-      libraries ← exerciseOps.getLibraries.map(ExercisesService.reorderLibraries(topLibraries, _))
-      user ← userOps.getUserByLogin(request.session.get("user").getOrElse(""))
-      progress ← userProgressOps.fetchMaybeUserProgress(user)
-    } yield (libraries, user, request.session.get("oauth-token"), progress, authorize)
-
-    for {
-      repo ← scalaexercisesRepo
-      result ← ops.runFuture map {
-        case Right((libraries, user, Some(token), progress, _))  ⇒ Ok(views.html.templates.home.index(user = user, libraries = libraries, progress = progress, repo = repo))
-        case Right((libraries, None, None, progress, authorize)) ⇒ Ok(views.html.templates.home.index(user = None, libraries = libraries, progress = progress, redirectUrl = Option(authorize.url), repo = repo)).withSession("oauth-state" → authorize.state)
-        case Right((libraries, Some(user), None, _, _))          ⇒ Unauthorized("Session token not found")
+  def index =
+    Secure(Action.async { implicit request ⇒
+      val ops = for {
+        authorize ← githubOps.getAuthorizeUrl(ConfigUtils.githubAuthId, ConfigUtils.callbackUrl)
+        libraries ← exerciseOps.getLibraries.map(
+          ExercisesService.reorderLibraries(topLibraries, _))
+        user     ← userOps.getUserByLogin(request.session.get("user").getOrElse(""))
+        progress ← userProgressOps.fetchMaybeUserProgress(user)
+      } yield (libraries, user, request.session.get("oauth-token"), progress, authorize)
+
+      for {
+        repo ← scalaexercisesRepo
+        result ← ops.runFuture map {
```

Comment:
we don't need to run these futures anymore with our new Play integration in Freestyle. Free prgrams can be returned directly in endpoints and we should not eagerly run futures. 
https://github.com/47deg/freestyle/blob/master/http/play/src/test/scala/PlayTests.scala#L67
https://github.com/47deg/freestyle/blob/master/http/play/src/main/scala/play.scala#L29